test/t/unit/test_unit_xfunc_scp_remote_files.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from conftest import assert_bash_exec, bash_env_saved
+
+
+@pytest.mark.bashcomp(cmd="ssh", ignore_env=r"^\+COMPREPLY=")
+class TestUnitXfuncScpRemoteFiles:
+    @pytest.fixture(scope="class")
+    def ssh_path(self, request, bash):
+        with bash_env_saved(bash) as bash_env:
+            bash_env.write_variable(
+                "PATH",
+                "$PWD/_comp_compgen_xfunc_scp_remote_files/bin:$PATH",
+                quote=False,
+            )
+            yield
+
+    def test_1(self, bash, ssh_path):
+        completions = (
+            assert_bash_exec(
+                bash,
+                r'cur=local:shared/default/ _comp_compgen -ax scp remote_files; printf "%s\n" "${COMPREPLY[@]}"; unset cur',
```

Comment:
It's actually better to use `bash_env` also for `cur`. That also includes the check that `_comp_xfunc_scp_cmpgen_remote_files` doesn't break the existing `cur`.

Reply:
I changed it to use `bash_env_saved` also in the test itself, is that what you meant?

Reply:
Thanks.


> I changed it to use `bash_env_saved` also in the test itself, is that what you meant?

Yes.